staging/src/k8s.io/apimachinery/pkg/util/proxy/upgradeaware.go

```diff
@@ -447,7 +447,19 @@ func (h *UpgradeAwareHandler) tryUpgrade(w http.ResponseWriter, req *http.Reques
 	// clean up the other half of the connection.
 	select {
 	case <-writerComplete:
+		// Once the client -> backend writer completes, we don't want to block
+		// forever, but also don't want to race in the case of network blips.
+		deadline := time.Now().Add(10 * time.Second)
```

Comment:
10 seconds is a lot, I did implement it in containerd just with 0.5 seconds and it seems to work fine 

https://github.com/containerd/containerd/commit/11a78d9d0f1664466fa3fffebd8ff234f3ef2677

Reply:
Yeah I didn't really know what to set as a reasonable value here. Thanks for the pointer, adopting 500ms.